menas/src/main/scala/za/co/absa/enceladus/menas/controllers/SchemaController.scala

```diff
@@ -36,18 +38,21 @@ import za.co.absa.enceladus.menas.utils.converters.SparkMenasSchemaConvertor
 import za.co.absa.enceladus.model.Schema
 import za.co.absa.enceladus.model.menas._
 
+import scala.util.control.NonFatal
 
+// TODO: use enumeration instead to type-force the schema types?
```

Comment:
Good idea! Enumeration or sealed case objects can improve things here.
https://pedrorijo.com/blog/scala-enums/

Reply:
I agree, good idea! Should improve our adherence to the open-closed principle (if we abstract the pattern-matching and use polymorphism to make a call to `SchemaParser::parse` instead of keeping the pattern matching here and callng concrete methods like `parseStructType` and `parseCopybook`)

Reply:
Related issue created: https://github.com/AbsaOSS/enceladus/issues/1228